src/main/java/Niwa.java

```diff
@@ -0,0 +1,322 @@
+
+    /** Variable to check if the chatbot is running */
+    private boolean isRunning;
+
+    /** List to store tasks */
+    private List<Task> tasks = new ArrayList<>();
+
+    /**
+     * Getter for the chatbot's name.
+     *
+     * @return The name of the chatbot.
+     */
+    public static String getName() {
+        return NAME;
+    }
+
+    /**
+     * Getter for the chatbot's logo.
+     *
+     * @return The logo of the chatbot.
+     */
+    public static String getLogo() {
+        return LOGO;
+    }
+
+    /**
+     * Getter for the isRunning flag.
+     *
+     * @return True if the chatbot is running, false otherwise.
+     */
+    public boolean getRunning() {
+        return isRunning;
+    }
+
+    /**
+     * Constructor initializes the chatbot and displays a greeting and user guide.
+     */
+    public Niwa() {
+        isRunning = true;
+        printGreet(NAME, LOGO);
+        processCommand(COMMAND_HELP_WORD);
+    }
+
+    /**
+     * Processes the command entered by the user.
+     *
+     * @param command The user command to process.
+     */
+    public void processCommand(String command) {
+        System.out.println(SEPARATOR);
+
+        command = command.trim();
+        String[] commandParts = command.split(" ", 2);
+
+        if (commandParts.length == 1) { // Handle single-word commands
+            switch (commandParts[0]) {
+            case COMMAND_BYE_WORD:
+                printExit();
+                isRunning = false;
+                break;
+            case COMMAND_LIST_WORD:
+                list();
+                break;
+            case COMMAND_HELP_WORD:
+                help();
+                break;
+            default:
+                echo(command);
+                break;
+            }
+        } else if (commandParts.length == 2) { // Handle commands with arguments
+            switch (commandParts[0]) {
+            case COMMAND_MARK_WORD:
+                mark(commandParts[1]);
+                break;
+            case COMMAND_UNMARK_WORD:
+                unmark(commandParts[1]);
+                break;
+            case COMMAND_DELETE_WORD:
+                delete(commandParts[1]);
+                break;
+            case COMMAND_EVENT_WORD:
+                // Fall through
+            case COMMAND_DEADLINE_WORD:
+                // Fall through
+            case COMMAND_TODO_WORD:
+                add(commandParts[1], commandParts[0]);
+                break;
+            default:
+                echo(command);
+                break;
+            }
+        }
+        System.out.println(SEPARATOR);
+    }
+
+    /**
+     * Lists all tasks in the task list.
+     */
+    private void list() {
+        System.out.println(PREFIX + COMMAND_LIST_SUCCESS);
+        int index = 1;
+
+        for (Task task : tasks) {
+            System.out.printf(PREFIX + "%d. %s%n", index++, task.getFullInfo());
```

Comment:
Unsure about this, but is it good to have index++ inside of the print statement? May make it hard to find if it causes an error.